src/ray/object_manager/plasma/dlmalloc.cc

```diff
@@ -133,6 +137,12 @@ void create_and_mmap_buffer(int64_t size, void **pointer, int *fd) {
 #endif
 
 void *fake_mmap(size_t size) {
+  if (allocated_once) {
+    RAY_LOG(DEBUG) << "fake_mmap called once already, refusing to allocate: " << size;
+    return MFAIL;
+  }
+  allocated_once = true;
```

Comment:
Shouldn't this only be set if `RayConfig::instance().preallocate_plasma_memory()` is set? Otherwise, won't every object that's created after the first object fail? Given that the tests are passing, I'm probably missing something here.

Reply:
Plasma is always allocating "object_store_size" memory on the first allocation, the preallocation flag only controls whether we are telling the OS to pre-allocate the pages of the file (vs leaving them as allocate-on-write pages).

Reply:
AH, that's right, thanks!